vars/folioJiraFlow.groovy

```diff
@@ -0,0 +1,147 @@
+
+void addJiraComment(String ticketNumber, String comment) {
+  String commentUrl = Constants.FOLIO_JIRA_ISSUE_URL + "${ticketNumber}/comment"
+  withCredentials([string(credentialsId: 'JiraFlow', variable: 'JiraToken')]) {
+    Map headers = [
+      "Content-Type" : "application/json",
+      "Authorization": "Bearer ${env.JiraToken}"
+    ]
+    String body = """
+      body: ${comment}
+      """
+    def comment_body = new JsonSlurperClassic().parseText(body)
+    new RestClient(this).post(commentUrl, headers, comment_body)
+  }
+}
+
+void addLabelJiraTicket(String ticketNumber, List labels) {
+  String labelUrl = Constants.FOLIO_JIRA_ISSUE_URL + "${ticketNumber}"
+  withCredentials([string(credentialsId: 'JiraFlow', variable: 'JiraToken')]) {
+    Map headers = [
+      "Content-Type" : "application/json",
+      "Authorization": "Bearer ${env.JiraToken}"
+    ]
+    labels.each { label ->
+      String body = """
+      { "update": { "labels": [{ "add" : "${label}"}]}}
+      """
+      def body_json = new JsonSlurperClassic().parseText(body)
+      new RestClient(this).put(labelUrl, headers, body_json)
+    }
+  }
+}
+
+void moveJiraTicket(List ticketNumbers, String moveId){
+  withCredentials([string(credentialsId: 'JiraFlow', variable: 'JiraToken')]) {
+    Map headers = [
+      "Content-Type" : "application/json",
+      "Authorization": "Bearer ${env.JiraToken}"
+    ]
+    ticketNumbers.each { ticket ->
+      String labelUrl = Constants.FOLIO_JIRA_ISSUE_URL + "${ticket}/transitions"
+      String body = """
+      {"transition": {"id": "${moveId}"}}
+      """
+      def body_json = new JsonSlurperClassic().parseText(body)
+      new RestClient(this).put(labelUrl, headers, body_json)
+    }
+  }
+}
+
+void processJiraTicketStatus(String summary) {
+  new Tools(this).copyResourceFileToWorkspace('jiraFlow/teamsInfo.json')
+  def teamsInfo = new JsonSlurperClassic().parseText(readJSON file: "teamsInfo.json")
+  withCredentials([string(credentialsId: 'JiraFlow', variable: 'JiraToken')]) {
+    Map headers = [
+      "Content-Type" : "application/json",
+      "Authorization": "Bearer ${env.JiraToken}"
+    ]
+//Below map is written for future use. PLEASE DO NOT DELETE!
+    Map status = [
+      "Open"          : 81,
+      "Closed"        : 61,
+      "Blocked"       : 71,
+      "In progress"   : 21,
+      "In review"     : 31,
+      "In code review": 91
+    ]
+    String updateUrl = Constants.FOLIO_JIRA_ISSUE_URL + "${ticket}/transitions"
+    teamsInfo.each { -> team
+      String body
+      def ticketNumbers = new JsonSlurperClassic().parseText(searchForExistingJiraTickets(jiraFlowJQL.getOpenTickets(summary, "${team.keySet().first()}")))
+      if (ticketNumbers["total"] != 0) {
+        ticketNumbers["issues"].each { ticket ->
+          switch (ticket["fields"]["status"]["name"]) {
+            case "Open":
+              new Common(this, "https://fakeUrl").logger.info("Similar ticket already exists, summary: ${summary}")
```

Comment:
It is better to initialize the Common class at the start of the function
Also if you need a logger, is better to initialize the Logger class from the utilities package.

Reply:
Thanks Sasha, indeed, instead tons of initializations inside loop, better to initialize it once.